packages/sync-repo-settings/src/sync-repo-settings.ts

```diff
@@ -146,21 +146,34 @@ async function updateMasterBranchProtection(repo: Repo, context: Context) {
       checks = customConfig.requiredStatusChecks;
     }
   }
-  await context.github.repos.updateBranchProtection({
-    branch: 'master',
-    owner,
-    repo: name,
-    required_pull_request_reviews: {
-      dismiss_stale_reviews: false,
-      require_code_owner_reviews: false,
-    },
-    required_status_checks: {
-      contexts: checks,
-      strict: config.requireUpToDateBranch,
-    },
-    enforce_admins: true,
-    restrictions: null!,
-  });
+  try {
+    await context.github.repos.updateBranchProtection({
+      branch: 'master',
+      owner,
+      repo: name,
+      required_pull_request_reviews: {
+        dismiss_stale_reviews: false,
+        require_code_owner_reviews: false,
+      },
+      required_status_checks: {
+        contexts: checks,
+        strict: config.requireUpToDateBranch,
+      },
+      enforce_admins: true,
+      restrictions: null!,
+    });
+  } catch (err) {
+    if (err.status === 401) {
```

Comment:
Just as a heads up, I tried this with merge-on-green as well (catching a specific error). Tyler and I were working on surfacing a specific error message when the bot didn't have access to push to a repository. It seemed like github at times was reusing error codes? In any case, it made me lose confidence that all error codes related to all errors in a 1:1 relationship.